src/OnboardingSPA/components/Drawer/DrawerPanel/DesignHeaderMenu.js

```diff
@@ -88,6 +89,36 @@ const DesignHeaderMenu = () => {
 		}
 	}, [ themeStatus ] );
 
+	useEffect( () => {
+		/* choose the first pattern if the fallback pattern is replaced with priamry/secondary type */
+		if (
+			patterns?.length > 0 &&
+			! patterns.some( ( pattern ) => pattern.slug === selectedPattern )
+		) {
+			const firstPattern = patterns[ 0 ];
+			updateSelectedPattern( firstPattern );
+		}
+	}, [ patterns, selectedPattern ] );
+
+	const updateSelectedPattern = ( pattern ) => {
+		setSelectedPattern( pattern.slug );
+		currentData.data.partHeader = pattern.slug;
+		setCurrentOnboardingData( currentData );
+
+		const newPagePattern = pattern.content + headerMenuPreviewData.pageBody;
+		setHeaderMenuData( newPagePattern );
+
+		setFlow( currentData ).then( ( result ) => {
+			if ( ! result?.error ) {
+				setCurrentOnboardingData( currentData );
+			}
+		} );
+
+		trackOnboardingEvent(
+			new OnboardingEvent( ACTION_HEADER_SELECTED, pattern )
+		);
```

Comment:
Do we have to send this Event? Is this a user action or something we do by default?